drivers/lhal/src/bflb_ef_ctrl.c

```diff
@@ -30,17 +40,23 @@
 #define EF_CTRL_EFUSE_R0_SIZE 128
 #endif
 
-#ifndef BOOTROM
-#define EF_CTRL_LOAD_BEFORE_READ_R0 bflb_ef_ctrl_load_efuse_r0(dev)
-#define EF_CTRL_LOAD_BEFORE_READ_R1 bflb_ef_ctrl_load_efuse_r1(dev)
+#ifndef BFLB_USE_ROM_DRIVER
```

Comment:
efuse can not use romdriver.

Reply:
Are you sure? As I could see, original bl_mcu_sdk (look at the V1.4.5 Makefile) uses romdriver by default:
```
option config to use
SUPPORT_ROMAPI?=y
... skipped to be short ...
cmake_definition+= -DCONFIG_ROMAPI=$(SUPPORT_ROMAPI)
```

and drivers/bl702_driver/CMakeLists.txt has following lines of code:
```
#add components denpend on this component
if(CONFIG_ROMAPI)
list(APPEND ADD_DEFINITIONS -DBFLB_USE_ROM_DRIVER)
endif()
```
I've tested my implementation with romdriver and it works perfectly, no issues were detected. EFUSE values were read properly and read results are valid especially for BLE MAC address values, which are exactly the same as the values, which were read by applications build with V1.4.5 branch of bl_mcu_sdk.

Reply:
> Are you sure? As I could see, original bl_mcu_sdk (look at the V1.4.5 Makefile) uses romdriver by default:
> 
> ```
> option config to use
> SUPPORT_ROMAPI?=y
> ... skipped to be short ...
> cmake_definition+= -DCONFIG_ROMAPI=$(SUPPORT_ROMAPI)
> ```
> 
> and drivers/bl702_driver/CMakeLists.txt has following lines of code:
> 
> ```
> #add components denpend on this component
> if(CONFIG_ROMAPI)
> list(APPEND ADD_DEFINITIONS -DBFLB_USE_ROM_DRIVER)
> endif()
> ```
> 
> I've tested my implementation with romdriver and it works perfectly, no issues were detected. EFUSE values were read properly and read results are valid especially for BLE MAC address values, which are exactly the same as the values, which were read by applications build with V1.4.5 branch of bl_mcu_sdk.

sure, this is new sdk,in order to make driver common, give up efuse rom driver

Reply:
Common driver is good reason, but ROM driver helps to save space in flash and I suppose it should be used as much as possible. At the same time there is no open documentation for EFUSE controller registers in provided BL702 reference manual and how could we could trust/verify/modify any EFUSE driver implementation? So using of the ROM driver seems to be more reliable decision in case of EFUSE. Provided patch allows user to choose what driver to use: ROM or SDK. And if in some future time we'll get some issues with ROM drivers we could safely turn to use SDK driver without any modifications of application code. This is common and generic way, already used in SDK for many driver API implementations. But what is your proposal regarding provided patch?